### PR TITLE
Add categories tabview

### PR DESCRIPTION
**Pull request by SunnyNagam**
Added categories to the front end by swipe-able tabs

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -27,20 +27,23 @@ dependencies {
...
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
+    implementation "org.jetbrains.kotlin:kotlin-script-runtime:$kotlin_version"
 
     // AndroidX
     implementation 'androidx.appcompat:appcompat:1.0.2'
     implementation 'androidx.browser:browser:1.0.0'
     implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
     implementation 'androidx.recyclerview:recyclerview:1.0.0'
     implementation 'androidx.cardview:cardview:1.0.0'
+    implementation 'com.google.android.material:material:1.1.0-alpha05'
```
Comment (brian-norman) [inline app/build.gradle:41]: Not an androidX change?

File: app/build.gradle [hunk 2]
```diff
@@ -27,20 +27,23 @@ dependencies {
...
 
     // Retrofit
     implementation 'com.squareup.retrofit2:retrofit:2.5.0'
     implementation 'com.squareup.retrofit2:converter-gson:2.5.0'
 
     // Glide
     implementation 'com.github.bumptech.glide:glide:4.9.0'
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
Comment (brian-norman) [inline app/build.gradle]: Not a Glide library

File: app/src/main/java/com/cobresun/factfulnewsandroid/TabFragment.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import timber.log.Timber
+
+
+// TODO: Rename parameter arguments, choose names that match
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
+private const val ARG_PARAM2 = "param2"
```
Comment (brian-norman) [inline app/src/main/java/com/cobresun/factfulnewsandroid/TabFragment.kt]: What are theeeeeese

File: app/src/main/java/com/cobresun/factfulnewsandroid/TabsPagerAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    FragmentStatePagerAdapter(fragmentManager) {
+
+
+    override fun getItem(position: Int): Fragment {
+        return TabFragment(position)
+    }
+
+
+    override fun getCount(): Int {
+        return 8                    //TODO replace with variable taking number of pages
```
Comment (brian-norman) [inline app/src/main/java/com/cobresun/factfulnewsandroid/TabsPagerAdapter.kt]: Don't hardcode

File: app/src/main/java/com/cobresun/factfulnewsandroid/ui/activities/MainActivity.kt [hunk 5]
```diff
@@ -4,14 +4,21 @@ import android.net.Uri
 import android.os.Bundle
 import android.view.View
 import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.Toolbar
 import androidx.browser.customtabs.CustomTabsIntent
 import androidx.core.content.res.ResourcesCompat
 import androidx.recyclerview.widget.LinearLayoutManager
+//import androidx.viewpager.widget.PagerAdapter
```
Comment (brian-norman) [inline app/src/main/java/com/cobresun/factfulnewsandroid/ui/activities/MainActivity.kt]: Can remove commented

File: app/src/main/java/com/cobresun/factfulnewsandroid/ui/activities/MainActivity.kt [hunk 6]
```diff
@@ -4,14 +4,21 @@ import android.net.Uri
...
+import androidx.viewpager.widget.ViewPager
+import com.cobresun.factfulnewsandroid.CategoryUtils
 import com.cobresun.factfulnewsandroid.R
 import com.cobresun.factfulnewsandroid.backend.api.ApiService
 import com.cobresun.factfulnewsandroid.backend.api.FetchResponse
 import com.cobresun.factfulnewsandroid.backend.models.Article
+import com.cobresun.factfulnewsandroid.TabsPagerAdapter
 import com.cobresun.factfulnewsandroid.ui.adapters.ArticlesAdapter
+import com.google.android.material.tabs.TabLayout
+//import com.google.android.material.tabs.TabLayout
```
Comment (brian-norman) [inline app/src/main/java/com/cobresun/factfulnewsandroid/ui/activities/MainActivity.kt]: Another commented

File: app/src/main/java/com/cobresun/factfulnewsandroid/CategoryUtils.kt [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.cobresun.factfulnewsandroid
+
+object CategoryUtils {
+    const val numberCategories = 8
+
+    val categoryFromTabIndex = arrayOf(
```
Comment (brian-norman) [inline app/src/main/java/com/cobresun/factfulnewsandroid/CategoryUtils.kt]: This array name doesn't make sense

File: app/src/main/java/com/cobresun/factfulnewsandroid/ui/activities/MainActivity.kt [hunk 8]
```diff
@@ -25,47 +32,63 @@ class MainActivity : AppCompatActivity() {
...
-        recyclerView.apply {
-            layoutManager = LinearLayoutManager(this@MainActivity)
-            adapter =
-                ArticlesAdapter(this@MainActivity, articles, itemOnClick)
-        }
+            override fun onTabReselected(tab: TabLayout.Tab) {}
+        })
     }
+
+//    private fun showArticles(articles: List<Article>) {
```
Comment (brian-norman) [inline app/src/main/java/com/cobresun/factfulnewsandroid/ui/activities/MainActivity.kt]: Can remove all this commented code?? LOL can you clean up ur PRs before getting them reviewed !!

File: app/src/main/res/layout/tab_fragment.xml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".All">
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recyclerView"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        app:layout_constraintBottom_toBottomOf="parent"
```
Comment (brian-norman) [inline app/src/main/res/layout/tab_fragment.xml]: Probably don't need constraints in this RecyclerView, for that matter this whole view might be able to just be one RecyclerView without the RelativeLayout ?

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -3,4 +3,7 @@
     <string name="article_photo_content_desc">An image from the article</string>
     <string name="article_menu_desc">Article more menu</string>
     <string name="action_share_title">Share</string>
+
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (brian-norman) [inline app/src/main/res/values/strings.xml]: Remove

File: app/src/main/res/values/styles.xml [hunk 11]
```diff
@@ -6,6 +6,9 @@
         <item name="colorPrimary">@color/colorPrimary</item>
         <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
         <item name="colorAccent">@color/colorAccent</item>
+
+        <item name="windowActionBar">false</item>
+        <item name="windowNoTitle">true</item>
```
Comment (brian-norman) [inline app/src/main/res/values/styles.xml:11]: These aren't used and can be deleted I think

**Review by brian-norman — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/styles.xml (hunk 11 above)
Comment (SunnyNagam) [reply app/src/main/res/values/styles.xml:11]: Nah it's used behind the scenes to make the toolbar run

**Review by brian-norman — APPROVED** [review]

**Pull request merged**